### PR TITLE
Don't deploy `PSP`s when `PodSecurityPolicy` plugin is disabled 

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
`PodSecurityPolicy` is deprecated and will be disabled in kubernetes v1.25+.
End-users are provided an option to migrate their PSPs before upgrading and disable the `PodSecurityPolicy` admission plugin in the ShootSpec. 
In that case, we should stop deploying our PSPs as well.

- This PR adapts the mutating fields of the existing PSPs in the PodSpec. 
- Stop deploying PSPs if the plugin is disabled in the Shoot.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/5250

**Special notes for your reviewer**:
/hold
Depends on https://github.com/gardener/gardener-extension-networking-calico/pull/201

**Release note**:

```breaking operator
Please make sure you're running gardener@v1.52 or above before upgrading to this version.
```

### CODE REVIEW

File: charts/internal/calico/templates/node-cpva/deployment-calico-node-vertical-autoscaler.yaml [hunk 1]
```diff
@@ -28,7 +28,11 @@ spec:
       {{- end }}
       priorityClassName: system-cluster-critical
       securityContext:
+        runAsNonRoot: true
         runAsUser: 65534
+        fsGroup: 1
+        supplementalGroups:
+        - 1
```
Comment (ScheererJ) [inline charts/internal/calico/templates/node-cpva/deployment-calico-node-vertical-autoscaler.yaml:35]: Why did you change this as well?
Setting `runAsNonRoot` seems a bit redundant, but will not hurt. Hence, I have no problem with it.
However, I do not get why you add `fsGroup` and `supplementalGroups`.

File: charts/internal/calico/templates/node/psp-calico.yaml [hunk 2]
```diff
@@ -36,3 +37,4 @@ spec:
   fsGroup:
     rule: RunAsAny
   readOnlyRootFilesystem: false
+{{- end}}
```
Comment (ScheererJ) [inline charts/internal/calico/templates/node/psp-calico.yaml:40]: Nit:
```suggestion
{{- end }}
```

File: charts/internal/calico/templates/typha-cpva/deployment-calico-typha-vertical-autoscaler.yaml [hunk 3]
```diff
@@ -33,6 +33,10 @@ spec:
       dnsPolicy: Default
       securityContext:
         runAsUser: 65534
+        runAsNonRoot: true
+        supplementalGroups:
+        - 1
+        fsGroup: 1
```
Comment (ScheererJ) [inline charts/internal/calico/templates/typha-cpva/deployment-calico-typha-vertical-autoscaler.yaml:39]: See above: Why did you add these?

File: charts/internal/calico/templates/typha/deployment-calico-typha.yaml [hunk 4]
```diff
@@ -75,6 +75,8 @@ spec:
       securityContext:
         runAsUser: 65534
         fsGroup: 65534
+        supplementalGroups:
+        - 1
```
Comment (ScheererJ) [inline charts/internal/calico/templates/typha/deployment-calico-typha.yaml:79]: See above: Why did you add this?

**Review by ScheererJ — CHANGES_REQUESTED** [review]
It might have been helpful to add the fact that this revendors `gardener` to v1.52.

Apart from that, I am not sure I understand some of the changes.

**Comment by shafeeqes** [comment]
> It might have been helpful to add the fact that this revendors `gardener` to v1.52.

I had opened https://github.com/gardener/gardener-extension-networking-calico/pull/201 for it and mentioned in the description of this PR as well.

File: charts/internal/calico/templates/node-cpva/deployment-calico-node-vertical-autoscaler.yaml (hunk 1 above)
Comment (shafeeqes) [reply charts/internal/calico/templates/node-cpva/deployment-calico-node-vertical-autoscaler.yaml:35]: These were part of the PSP under `MustRunAs`, but there it was actually a specified range, so I looked at the actual components running, and found these values there. Aren't these required? Can we just drop these fields? How does it work when the PSP is not present anymore?

**Comment by ScheererJ** [comment]
> > It might have been helpful to add the fact that this revendors `gardener` to v1.52.
> 
> I had opened #201 for it and mentioned in the description of this PR as well.

My bad. I overlooked that this depends on #201 according to the pull request description. Therefore, I was wondering why it was not mentioned.

File: charts/internal/calico/templates/node-cpva/deployment-calico-node-vertical-autoscaler.yaml (hunk 1 above)
Comment (ScheererJ) [reply charts/internal/calico/templates/node-cpva/deployment-calico-node-vertical-autoscaler.yaml:35]: My understanding of podsecuritypolicy was off. Your change is good.

**Comment by shafeeqes** [comment]
Once #201 is merged, I'll rebase this PR.

**Review by ScheererJ — DISMISSED** [review]
/lgtm

**Comment by ScheererJ** [comment]
@shafeeqes Please rebase.

**Comment by ScheererJ** [comment]
> > @shafeeqes Please rebase.
> 
> Done.

I am confused. It still looks like the revendoring is part of this pull request even though it should not be. Can you please take a look again?

**Comment by shafeeqes** [comment]
> > > @shafeeqes Please rebase.
> > 
> > 
> > Done.
> 
> I am confused. It still looks like the revendoring is part of this pull request even though it should not be. Can you please take a look again?

I am really sorry, I don't know why it's happening again and again. It did skip these commits when I rebased. 
I will take extra care next time and confirm.

**Review by ScheererJ — APPROVED** [review]
/lgtm
